### PR TITLE
[FLOSS H2] doc: fix wording in DESIGN.md

### PR DESCRIPTION
**Pull request by lawli3t**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Review by markus2330 — CHANGES_REQUESTED** [review]
The content of all PRs are okay but in all of them the release notes were not yet updated.

**Comment by lawli3t** [comment]
> The content of all PRs are okay but in all of them the release notes were not yet updated.
Sorry, it completely slipped my mind - probably because i was hurrying a bit too much

**Comment by muskater** [comment]
Good work, I think this PR improves the syntactic correctness and the wording of this important document. While we are at it, you might want to consider to include the following two errors:

 
Line 9:
Elektra [aims](GOALS.md) at following design principles:
-1. To make the API **futureproof** so that it can remain compatible and stable
+1. To make the API **future-proof** so that it can remain compatible and stable
    over a long period of time,
 2. to make it hard to use the API the wrong way by making it simple&robust, and
 3. to make the API easy to use for programmers reading and writing

Line 43: 
The second function may also shut
-down connections. **Therefore** it must be called before the end of a program.
+down connections. **Therefore,** it must be called before the end of a program.

File: doc/DESIGN.md [hunk 1]
```diff
@@ -4,11 +4,11 @@ This document describes the design of Elektra's C-API and provides hints for
 binding writers. It is not aimed at plugin writers, since it does not
 talk about the implementation details of Elektra.
 
-Elektra [aims](GOALS.md) at following design principles:
+Elektra [aims](GOALS.md) to fulfill the following design principles:
 
 1. To make the API futureproof so that it can remain compatible and stable
```
Comment (muskater) [inline doc/DESIGN.md:9]: My suggestions are:
Line 9: future-proof
Line 43: Therefore,

**Review by muskater — APPROVED** [review]
Good work, I think this PR improves the syntactic correctness and the wording of this important document. While we are at it, you might want to consider to include the following two errors:

Line 9:
Elektra aims at following design principles:
-1. To make the API futureproof so that it can remain compatible and stable
+1. To make the API future-proof so that it can remain compatible and stable
over a long period of time,
2. to make it hard to use the API the wrong way by making it simple&robust, and
3. to make the API easy to use for programmers reading and writing

Line 43:
The second function may also shut
-down connections. Therefore it must be called before the end of a program.
+down connections. Therefore, it must be called before the end of a program.

**Review by JakobWonisch — APPROVED** [review]
Good changes! The release notes have also now been update, so it looks good to go!

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
There are a still a few CI errors.

**Comment by markus2330** [comment]
Thank you for the improvements!

**Pull request merged**